packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -464,43 +464,48 @@ class AnimationController extends Animation<double>
-  ///
-  /// If [from] is non-null, it will be set as the current [value] before running
-  /// the animation.
-  ///
-  /// The most recently returned [TickerFuture], if any, is marked as having been
-  /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
-  /// derivative future completes with a [TickerCanceled] error.
-  ///
-  /// During the animation, [status] is reported as [AnimationStatus.forward],
-  /// which switches to [AnimationStatus.completed] when [upperBound] is
-  /// reached at the end of the animation.
-  TickerFuture forward({ double? from }) {
+  TickerFuture _toggle(String method, bool animateForward, double? from) {
     assert(() {
-      if (duration == null) {
+      if (duration == null && (animateForward || reverseDuration == null)) {
+        final (String duration, String durationQuotes) = animateForward
+            ? ('duration', '"duration"')
+            : ('duration or reverseDuration', '"duration" or "reverseDuration"');
         throw FlutterError(
-          'AnimationController.forward() called with no default duration.\n'
-          'The "duration" property should be set, either in the constructor or later, before '
-          'calling the forward() function.',
+          'AnimationController.$method() called with no default $duration.\n'
+          'The $durationQuotes property should be set, either in the constructor or later, before '
+          'calling the $method() function.',
         );
       }
       return true;
     }());
     assert(
       _ticker != null,
-      'AnimationController.forward() called after AnimationController.dispose()\n'
+      'AnimationController.$method() called after AnimationController.dispose()\n'
       'AnimationController methods should not be used after calling dispose.',
     );
-    _direction = _AnimationDirection.forward;
+    _direction = animateForward ? _AnimationDirection.forward : _AnimationDirection.reverse;
     if (from != null) {
       value = from;
     }
-    return _animateToInternal(upperBound);
+    return _animateToInternal(animateForward ? upperBound : lowerBound);
   }
 
+  /// Starts running this animation forwards (towards the end).
+  ///
+  /// Returns a [TickerFuture] that completes when the animation is complete.
+  ///
+  /// If [from] is non-null, it will be set as the current [value] before running
+  /// the animation.
+  ///
+  /// The most recently returned [TickerFuture], if any, is marked as having been
+  /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
+  /// derivative future completes with a [TickerCanceled] error.
+  ///
+  /// During the animation, [status] is reported as [AnimationStatus.forward],
+  /// which switches to [AnimationStatus.completed] when [upperBound] is
+  /// reached at the end of the animation.
+  TickerFuture forward({ double? from }) => _toggle('forward', true, from);
```

Comment:
Nit: Would it be more readable if the parameters to _toggle were named parameters?

Reply:
yeah, that sounds really good!